src/options.cpp

```diff
@@ -456,7 +500,81 @@ bool cOpt::operator!=(const std::string sCompare) const
     return !(*this == sCompare);
 }
 
-void initOptions()
+/** Fill TILESETS mapping with values.
+ * Scans all directores in gfx directory for file named "tileset.txt".
+ * All founded values added in mapping TILESETS as name, tileset_dir.
+ * Furthermore, it builds possible values list for cOpt class.
+ * @return One string containing all found tilesets in form "tileset1,tileset2,tileset3,..."
+ */
+static std::string build_tilesets_list()
+{
+    const std::string defaultTilesets = "hoder,deon";
+
+    std::string tileset_names;
+    bool first_tileset_name = true;
+
+    TILESETS.clear();
+
+    auto const tilesets_dirs = get_directories_with(FILENAMES["tileset-conf"], FILENAMES["gfxdir"], true);
+
+    for( auto &ts_dir : tilesets_dirs ) {
+        std::ifstream fin;
+        std::string file = ts_dir + "/" + FILENAMES["tileset-conf"];
+
+        fin.open( file.c_str() );
+        if(!fin.is_open()) {
+            fin.close();
```

Comment:
This is old behavior. Really i don't know why close() is calling. Will be fixed.